torch/_functorch/aot_autograd.py

```diff
@@ -1272,6 +1272,15 @@ class AOTConfig:
     num_params_buffers: int
     aot_id: int
     keep_inference_input_mutations: bool
+    # If None, defer to config
+    _dynamic_shapes: Optional[bool] = None
```

Comment:
Any reason not to just kill the functorch config completely, and base our decision completely off of whether the input fake tensors have a shape env?

Reply:
I have the same question here.

Reply:
I'm happy to do this in a follow up, there is just annoying test suite fixing that would need to happen